src/shared/libs/iota/utils.js

```diff
@@ -425,28 +425,44 @@ export const withRetriesOnDifferentNodes = (nodes, failureCallbacks) => {
  * Fetches list of IRI nodes from a server
  *
  * @method fetchRemoteNodes
- * @param {string} [url]
- * @param {object} [options]
  *
- * @returns {Promise<*>}
+ * @returns {Promise<array>}
  */
-export const fetchRemoteNodes = (
-    url = NODELIST_URL,
-    options = {
+export const fetchRemoteNodes = async () => {
+    const requestOptions = {
         headers: {
             Accept: 'application/json',
         },
-    },
-) =>
-    fetch(url, options)
-        .then((response) => response.json())
-        .then((response) => {
-            if (isArray(response)) {
-                return response.filter((node) => typeof node.node === 'string' && node.node.indexOf('https://') === 0);
+    };
+
+    let remoteNodes = [];
+
+    for (let index = 0; index < NODELIST_ENDPOINTS.length; index++) {
+        try {
+            const endPoint = NODELIST_ENDPOINTS[index];
+
+            const response = await fetch(endPoint, requestOptions, GET_NODELIST_REQUEST_TIMEOUT);
```

Comment:
`fetch` does not appear to support timeouts according to the [spec](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch). To achieve similar functionality we use `axios` on mobile:

https://github.com/iotaledger/trinity-wallet/blob/f3fab95c9f20a9afc21a36ecf8473ca664ae49b7/src/mobile/src/ui/routes/entry.js#L187-L208